Extensions/bezier/bezier.lisp

```diff
@@ -844,25 +844,110 @@ second curve point, yielding (200 50)."
 
 ;;; Fallback method (suitable for CLX)
 
-(defmethod medium-draw-bezier-design* (medium design)
+(defmethod medium-draw-bezier-design* (medium design options)
   (render-through-pixmap design medium))
 
 
+;;; The real CLX method
+
+(defun my-compute-rgb-image-pixmap (drawable image)
+  (let* ((width (climi::image-width image))
+         (height (climi::image-height image))
+         (depth (xlib:drawable-depth drawable))
+         (im (clim-clx::image-to-ximage-for-drawable drawable image)))
+    (setf width (max width 1))
+    (setf height (max height 1))
+    (let* ((pixmap (xlib:create-pixmap :drawable drawable
+				       :width width
+				       :height height
+				       :depth depth))
+	   (gc (xlib:create-gcontext :drawable pixmap)))
+      (unless (or (>= width 2048) (>= height 2048)) ;### CLX bug
+	(xlib:put-image pixmap gc im
+			:src-x 0 :src-y 0
+			:x 0 :y 0
+			:width width :height height))
+      (xlib:free-gcontext gc)
+      pixmap)))
+
+(defun my-compute-rgb-image-mask (drawable image)
+  (let* ((width (climi::image-width image))
+         (height (climi::image-height image))
+         (bitmap (xlib:create-pixmap :drawable drawable
+                                     :width width
+                                     :height height
+                                     :depth 1))
+         (gc (xlib:create-gcontext :drawable bitmap
+				   :foreground 1
+				   :background 0))
+         (idata (climi::image-data image))
+         (xdata (make-array (list height width)
+			    :element-type '(unsigned-byte 1)))
+         (im (xlib:create-image :width width
+                                :height height
+                                :depth 1
+                                :data xdata)) )
+    (dotimes (y width)
+      (dotimes (x height)
+        (if (> (aref idata x y) #x80000000)
+            (setf (aref xdata x y) 0)
+	    (setf (aref xdata x y) 1))
+        ;; (print (aref idata x y) *terminal-io*)
+        (when (= (aref idata x y) #x00FFFFFF)
+          (setf (aref xdata x y) 0))))
+    (unless (or (>= width 2048) (>= height 2048)) ;### CLX breaks here
```

Comment:
interesting. afair it butchered too big images into scanlines and send them one-by-one (put-image that is), so it means that such big image scanline exceeds one request limit.